src/chatClient.cpp

```diff
@@ -1825,20 +1851,45 @@ ChatRoomList::~ChatRoomList()
 
 GroupChatRoom::GroupChatRoom(ChatRoomList& parent, const mega::MegaTextChat& aChat)
 :ChatRoom(parent, aChat.getHandle(), true, aChat.getShard(),
-  (chatd::Priv)aChat.getOwnPrivilege(), aChat.getCreationTime()),
-  mHasTitle(false), mRoomGui(nullptr)
+  (chatd::Priv)aChat.getOwnPrivilege(), aChat.getCreationTime()), mRoomGui(nullptr)
 {
+    auto title = aChat.getTitle();
+    if (title && title[0])
+    {
+        mEncryptedTitle = title;
+        mHasTitle = true;
+    }
+
     auto peers = aChat.getPeerList();
     if (peers)
     {
+        std::vector<promise::Promise<void> > promises;
         auto size = peers->size();
+        auto wptr = weakHandle();
         for (int i=0; i<size; i++)
         {
             auto handle = peers->getPeerHandle(i);
             assert(handle != parent.client.myHandle());
             mPeers[handle] = new Member(*this, handle, (chatd::Priv)peers->getPeerPrivilege(i)); //may try to access mContactGui, but we have set it to nullptr, so it's ok
+            promises.push_back(mPeers[handle]->nameResolved());
         }
+
+        // If there is not any promise at vector promise, promise::when is resolved directly
+        mNameMemberResolved = promise::when(promises)
+        .then([wptr, this]()
+        {
+            wptr.throwIfDeleted();
+            if (!mHasTitle)
+            {
+                makeTitleFromMemberNames();
+            }
+        });
+    }
+    else
+    {
+        makeTitleFromMemberNames();
```

Comment:
Shouldn't we check here whether `mHasTitle` or not?
`if (!peers)` means we are alone in the groupchat, but it doesn't change the logic underneath.

Reply:
Yes, you are right. It's necessary add `mHasTitle`condition